lib/ActivityStreams.js

```diff
@@ -98,7 +98,8 @@ function ActivityStreams(options = {}) {
   );
 
   this._metadataStore = new MetadataStore();
-  this._previewProvider = new PreviewProvider(this._tabTracker);
+  this._metadataStore.asyncConnect();
```

Comment:
We can't do the auto-connection in the constructor is mainly due to the fact that the underlying `connect()` in Sqlite.jsm is an async function, the user might have issued some queries upon this connection before it actually gets connected. So we have to explicitly connect to the metadata store, wait until the connection is established, and then proceed queries upon it.

That being said, it's preferred to just create the MetadataStore object here, and call the "asyncConnect" in the PreviewProvider before calling any query APIs.